rareapi/views/Profiles.py

```diff
@@ -0,0 +1,45 @@
+from django.http.response import HttpResponseServerError
+from rest_framework.response import Response
+from rest_framework.viewsets import ViewSet
+from rest_framework.decorators import action
+from django.contrib.auth.models import User
+from rareapi.models import RareUsers
+from rest_framework import status
+from django.core.exceptions import ValidationError
+from rest_framework import serializers
+
+class ProfileViewSet(ViewSet):
+    def list(self, request):
+        users = RareUsers.objects.all()
+        serializer = BasicProfileSerializer(users, many=True, context={'request':request})
+        return Response(serializer.data)
+
+
+    """action to toggle between admin and author status depending on message from client"""
+    @action(methods=['patch'], detail=True)
+    def update_role(self, request, pk=None):
+        try:
+            rare_user = RareUsers.objects.get(pk=pk)
+            user = User.objects.get(pk=rare_user.id)
+        except RareUsers.DoesNotExist:
+            return Response({'message': 'user does not exit'}, status=status.HTTP_404_NOT_FOUND)
+        if not request.auth.user.is_staff:
+            return Response({'message':'only admins can change user roles'}, status=status.HTTP_403_FORBIDDEN)
+        if request.data["update_role"] == "author":
```

Comment:
I would have expected the server to be looking for a property called is_staff, with a value of true or false. What we have here works, and is functionally the same, just with different property and value names than I would have expected.